opensearch/chart/templates/users-secret.yaml

```diff
@@ -2,15 +2,20 @@
 # SPDX-License-Identifier: Apache-2.0
 
 {{- $labels := include "opensearch-cluster-local.labels" . }}
-{{- range .Values.cluster.users }}
+{{/*
+WARNING: defaultPasswords are dummy passwords for development only!
+MUST be overridden for production environments!
+*/}}
+{{- $defaultPasswords := dict "logs" "DUMMY-LogsUser2025!@#" "logs2" "DUMMY-FailoverUser2025$%^" }}
+{{- range $username, $credentials := .Values.cluster.usersCredentials }}
 ---
 apiVersion: v1
 kind: Secret
 metadata:
-  name: {{ .name }}-credentials
+  name: {{ $username }}-credentials
   labels: {{ $labels | nindent 4 }}
 type: Opaque
 data:
-  username: {{ .name | b64enc }}
-  password: {{ default (printf "%s%s" (randAlphaNum 15) (index (list "!" "@" "#" "$" "%" "^" "&" "*") (randInt 0 8))) .password | b64enc }}
+  username: {{ default $username $credentials.username | b64enc }}
+  password: {{ default (index $defaultPasswords $username) $credentials.password | b64enc }}
 {{- end }}
```

Comment:
I would suggest changing the key here to something other than `$username`. It is a bit strange to (on line 19) have this logic:

```
username: {{ default $username $credentials.username | b64enc }}
```
I think this is a bit confusing: `$credentials.username` is for the key username instead of the key `$username`:

Similar on line 20:
```
password: {{ default (index $defaultPasswords $username) $credentials.password | b64enc }}
```

Maybe you could call it `failoverUser`, `failoverObject` or something similar?